### PR TITLE
Anonymous users should not connect to the WS

### PR DESCRIPTION
**Pull request by kanat**
### 🎯 Goal

Align with other SDKs

#### General
- [ ] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#flutter-team) (required)
- [ ] PR is linked to the GitHub issue it resolves

### ☑️Reviewer Checklist
- [ ] Sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] All code we touched has new or updated Documentation

### CODE REVIEW

File: packages/stream_video/lib/src/stream_video.dart [hunk 1]
```diff
@@ -251,6 +251,10 @@ class StreamVideo {
 
   /// Connects the user to the Stream Video service.
   Future<Result<UserToken>> connect() async {
+    if (currentUserType == UserType.anonymous) {
+      _logger.w(() => '[connect] rejected (anonymous user)');
+      return Result.error('Cannot connect anonymous user to the WS');
```
Comment (Nash0x7E2) [inline packages/stream_video/lib/src/stream_video.dart:256]: Should this be considered an erorr, @kanat?

For example, if I am following the tutorial and decide to connect as a guest, an error here might be a deceiving.

Comment (kanat) [reply packages/stream_video/lib/src/stream_video.dart:256]: There are 3 user types:
- authenticated
- guest
- anonymous

For `anonymous` we should not connect to the WS.

This logic is common for all SDKs.
Here is the iOS [implementation](https://github.com/GetStream/stream-video-swift/blob/main/Sources/StreamVideo/StreamVideo.swift#L503-L507)

Comment (Nash0x7E2) [reply packages/stream_video/lib/src/stream_video.dart:256]: That's fair. Can we update the text there to be a bit more descriptive? 

iOS for example uses "Missing Permissions" as the error message.

Comment (kanat) [reply packages/stream_video/lib/src/stream_video.dart]: Updated 👍

**Review by Nash0x7E2 — APPROVED** [review]
LGTM

**Pull request merged**